### PR TITLE
Isolating Service Code

### PR DESCRIPTION
**Pull request by shreypdev**
# Overview

## Links to jira tickets

- {Include links to all of the applicable Jira tickets}

## This PR contains the following changes

- {List all the relevant changes. If there are many changes across many Jira tickets, organize the changes by Jira ticket}

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/models/project-view.ts [hunk 1]
```diff
@@ -5,6 +5,18 @@ import { getLogger } from '../utils/logger';
...
+export interface IGetProject {
+  id: number;
+  coordinator: GetCoordinatorData | null;
+  permit: any;
+  project: any;
+  objectives: GetObjectivesData | null;
+  location: GetLocationData | null;
+  iucn: GetIUCNClassificationData | null;
+  funding: GetFundingData | null;
+  partnerships: GetPartnershipsData | null;
```
Comment (NickPhura) [inline api/src/models/project-view.ts]: small nits: 
There are interfaces defined for permit and project that we can assign here.
Also, are these ever null? Its only used in 1 function and I think it always returns an object for each of these?

**Comment by NickPhura** [comment]
Found an issue with deleting project funding source:
- Created a project with 3 funding sources.
- Edited them a bit (all worked fine).
- Tried to delete one of them, but got an error saying `Failed to delete survey funding source`

(No surveys had been created when I tried the above)

If I then create a survey, and associated the funding sources to the survey, \I am then able to delete the funding sources.
I can't remember if that is correct or wrong though (ie: should funding sources be deletable if they are associated to a survey? Might have to check what the behaviour of SIMS is in dev currently)

File: api/src/models/project-view.ts (hunk 1 above)
Comment (shreypdev) [reply api/src/models/project-view.ts]: So I added types for Permit and Project. Nice catch! 
About the nulls, we need them as they are used in `getProjectEntitiesById` which only return entities that are requested and else others will be null.

**Review by NickPhura — APPROVED** [review]
Nice work!

**Pull request merged**